### PR TITLE
[Doc] Add assigner doc of rtmdet_description.md

### PR DESCRIPTION
**Pull request by satuoqaq**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDetection or MMClassification.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: docs/zh_cn/algorithm_descriptions/rtmdet_description.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+这些算法将 `预测的 Bboxes 与 GT 的 IOU ` 和 `分类分数`
+或者是对应 `分类Loss` 和 `回归Loss` 拿来计算 `Matching Cost` 矩阵再通过 `top-k`
+的方式动态决定样本选取以及样本个数。通过这种方式,
+在网络优化的过程中会自动选取对分类或者回归更加敏感有效的位置的样本,
+它不再只依赖先验的静态的信息, 而是使用当前的预测结果去动态寻找最优的匹配,
+只要模型的预测越准确, 匹配算法求得的结果也会更优秀。但是在网络训练的初期,
+网络的分类以及回归是随机初始化, 这个时候还是需要 `先验` 来约束, 以达到 `冷启动` 的效果。
+
+综上, `RTMDet` 采用 `Dynamic Soft Label Assigner` 来实现标签的动态匹配策略,该方法主要包括
+使用 **位置先验信息损失** , **样本回归损失** , **样本分类损失** , 同时对三个损失进行了`Soft`
```
Comment (PeterH0323) [inline docs/zh_cn/algorithm_descriptions/rtmdet_description.md:25]:
```suggestion
使用 **位置先验信息损失** , **样本回归损失** , **样本分类损失** , 同时对三个损失进行了 `Soft`
```

**Review by PeterH0323 — CHANGES_REQUESTED** [review]

File: docs/zh_cn/algorithm_descriptions/rtmdet_description.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
+### 标签匹配策略
+
+标签匹配策略 `Label Assignment` 是目标检测模型训练中最核心的问题之一,
+更好的样本选取往往能够使得网络更好学习到物体的特征以提高检测能力。
```
Comment (hhaAndroid) [inline docs/zh_cn/algorithm_descriptions/rtmdet_description.md:4]: 更好的标签匹配策略xx

File: docs/zh_cn/algorithm_descriptions/rtmdet_description.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+早期样本选取一般都是基于 `空间以及尺度信息的先验` 来决定样本的选取。
```
Comment (hhaAndroid) [inline docs/zh_cn/algorithm_descriptions/rtmdet_description.md:6]: 早期的样本匹配策略一般是xx

File: docs/zh_cn/algorithm_descriptions/rtmdet_description.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+`RetinaNet` 则是通过 `Anchor` 与 `GT` 的最大 `IOU` 匹配来划分正负样本;
+`YOLOV5` 的正负样本则是通过样本的宽高比先筛选一部分, 然后通过位置信息选取
+`GT` 中心落在的 `Grid` 以及临近的两个作为正样本。
+
+但是上述方法都是属于基于 `先验` 的静态匹配策略, 就是样本的选取方式是根据人的经验规定的。
+不会随着网络的优化而进行自动优化选取到更好的样本, 近些年涌现了许多优秀的动态标签匹配策略：
+如 `OTA` 提出使用 `Sinkhorn` 迭代求解匹配中的最优传输问题, `YOLOX` 中使用 `OTA` 的近似算法
+`SimOTA` , `TOOD` 将分类分数以及 `IOU` 相乘计算 `Cost` 矩阵进行标签匹配等等。
+这些算法将 `预测的 Bboxes 与 GT 的 IOU ` 和 `分类分数`
+或者是对应 `分类Loss` 和 `回归Loss` 拿来计算 `Matching Cost` 矩阵再通过 `top-k`
```
Comment (hhaAndroid) [inline docs/zh_cn/algorithm_descriptions/rtmdet_description.md:17]: 分类 Loss

File: docs/zh_cn/algorithm_descriptions/rtmdet_description.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+的方式动态决定样本选取以及样本个数。通过这种方式,
+在网络优化的过程中会自动选取对分类或者回归更加敏感有效的位置的样本,
+它不再只依赖先验的静态的信息, 而是使用当前的预测结果去动态寻找最优的匹配,
+只要模型的预测越准确, 匹配算法求得的结果也会更优秀。但是在网络训练的初期,
+网络的分类以及回归是随机初始化, 这个时候还是需要 `先验` 来约束, 以达到 `冷启动` 的效果。
+
+综上, `RTMDet` 采用 `Dynamic Soft Label Assigner` 来实现标签的动态匹配策略,
```
Comment (hhaAndroid) [inline docs/zh_cn/algorithm_descriptions/rtmdet_description.md:24]: 前面要补充一句： 

RTMDet 作者也是采用了动态的 SimOTA 做法，不过其对动态的正负样本分配策略进行了改进。

之前的动态分类策略（HungarianAssigner、OTA）往往使用与 Loss 完全一致的代价函数作为匹配的依据，但我们经过实验发现这并不一定时最优的。

使用更多 soften 的 cost 以及先验，能够提升性能。为此提出了 Dynamic Soft Label Assigner

File: docs/zh_cn/algorithm_descriptions/rtmdet_description.md [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+该方法主要包括使用
+**位置先验信息损失** , **样本回归损失** , **样本分类损失** , 同时对三个损失进行了 `Soft`
+处理进行参数调优, 以达到最佳的动态匹配效果。
```
Comment (hhaAndroid) [inline docs/zh_cn/algorithm_descriptions/rtmdet_description.md:27]: ppt 里面的图记得贴进去

File: docs/zh_cn/algorithm_descriptions/rtmdet_description.md [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+soft_cls_cost = soft_cls_cost.sum(dim=-1)
+```
+
+最终的 Matching Cost 矩阵由如下损失构成：
+
+```python
+cost_matrix = soft_cls_cost + iou_cost + soft_center_prior
+```
+
+再使用 `SimOTA` , 先通过每一个 `GT` 计算所有样本 `IOU的和` 决定每一个 `GT` 选择多少个样本,
```
Comment (hhaAndroid) [inline docs/zh_cn/algorithm_descriptions/rtmdet_description.md:78]: 这句话过于简单了，可以参考之前 YOLOX 解读来加一下内容

File: docs/zh_cn/algorithm_descriptions/rtmdet_description.md [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+soft_cls_cost = F.binary_cross_entropy_with_logits(
+    valid_pred_scores, soft_label,
+    reduction='none') * scale_factor.abs().pow(2.0)
+soft_cls_cost = soft_cls_cost.sum(dim=-1)
+```
+
+最终的 Matching Cost 矩阵由如下损失构成：
+
+```python
+cost_matrix = soft_cls_cost + iou_cost + soft_center_prior
```
Comment (hhaAndroid) [inline docs/zh_cn/algorithm_descriptions/rtmdet_description.md:75]: 公式要贴到前面原理部分
